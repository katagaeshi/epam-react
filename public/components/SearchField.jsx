import React from 'react';

const placeholder = 'Enter movie info here';

const SearchField = () => <input placeholder={placeholder} />;

export default SearchField;
