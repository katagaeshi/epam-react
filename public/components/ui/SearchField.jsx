import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

const styles = {
  searchField: {
    'grid-row': '2',
    'grid-column': '1 / 3',
  },
};

const placeholder = 'Enter movie info here';

class SearchField extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = props.handleChange;
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  handleKeyPress(event) {
    if (event.key === 'Enter') {
      this.props.onEnter();
    }
  }

  render() {
    return (<input
      className={this.props.classes.searchField}
      placeholder={placeholder}
      onChange={this.handleChange}
      onKeyPress={this.handleKeyPress}
    />);
  }
}

SearchField.propTypes = {
  handleChange: PropTypes.func.isRequired,
  onEnter: PropTypes.func.isRequired,
  classes: PropTypes.shape({
    searchField: PropTypes.string,
  }).isRequired,
};

export default injectSheet(styles)(SearchField);
