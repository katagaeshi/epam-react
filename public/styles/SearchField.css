.SearchField {
  display: block;
  width: 100%;
}
