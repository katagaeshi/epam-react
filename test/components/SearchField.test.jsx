import React from 'react';
import renderer from 'react-test-renderer';
import SearchField from './../../public/components/SearchField';

describe('SearchField', () => {
  it('Snapshot', () => {
    const tree = renderer
      .create(<SearchField />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
