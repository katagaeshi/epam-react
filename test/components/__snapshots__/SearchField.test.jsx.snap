// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField Snapshot 1`] = `
<input
  className="SearchField"
  placeholder="Enter movie info here"
/>
`;
